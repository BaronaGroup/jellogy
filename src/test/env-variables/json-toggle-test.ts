process.env.KARHU_JSON = 'true'
import {prepareTestKarhu} from '../test-utils'

describe('json-toggle-test', () => {
  const karhuTest = prepareTestKarhu({})

  it('the default config produces json if KARHU_JSON is set to a truthy value', karhuTest((karhu, output) => {
    karhu.context('test-context').error('Should be json', {value: true})
    expect(output.tracked).toMatchSnapshot()
  }))
})
