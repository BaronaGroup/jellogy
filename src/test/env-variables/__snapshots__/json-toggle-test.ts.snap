// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`json-toggle-test the default config produces json if KARHU_JSON is set to a truthy value 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [],
  "INFO" => Array [],
  "WARN" => Array [],
  "ERROR" => Array [
    Array [
      "{\\"timestamp\\":1,\\"context\\":\\"test-context\\",\\"logLevel\\":\\"ERROR\\",\\"details\\":[\\"Should be json\\",{\\"value\\":true}]}",
      "ERROR",
      "test-context",
    ],
  ],
}
`;
